sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubClient.cs

```diff
@@ -270,7 +282,103 @@ public virtual EventSender CreateSender(SenderOptions senderOptions = default)
         ///
         [EditorBrowsable(EditorBrowsableState.Never)]
         public override string ToString() => base.ToString();
+
+        /// <summary>
+        ///   Performs the actions needed to validate the <see cref="EventHubClientOptions" /> associated
+        ///   with this client.
+        /// </summary>
+        ///
+        /// <param name="clientOptions">The set of options to validate.</param>
+        ///
+        /// <remarks>
+        ///   In the case that the options violate an invariant or otherwise represent a combination that
+        ///   is not permissible, an appropriate exception will be thrown.
+        /// </remarks>
+        ///
+        protected virtual void ValidateClientOptions(EventHubClientOptions clientOptions)
```

Comment:
Does this have to be protected?

Reply:
In support of unit testing, yes, I believe there is value in allowing mocks to inherit and accept a potentially invalid set of options.

Reply:
Same sentiment as below;   I'll give it another look and reconsider the approach before the next batch of changes; I'm trying to strike the right balance between keeping things internal and making sure that our library is fully supportive of testing by consumers. One of the big flaws of many of the current libraries is that they're overly restrictive where testability is concerned.

Reply:
Revisited notes and scenarios around testing and decided that I'm in agreement with your thoughts;  I've made connection string parsing internal and allowed for child classes to set the client options directly, which gives the flexibility needed to avoid the connection string constructors.